security/cmd/chiron/main.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
+	istiocmd "istio.io/istio/pkg/cmd"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/security/pkg/cmd"
+	chiron "istio.io/istio/security/pkg/k8s/chiron"
+	"istio.io/pkg/collateral"
+	"istio.io/pkg/ctrlz"
+	"istio.io/pkg/log"
+	"istio.io/pkg/version"
+)
+
+var (
+	opts = cliOptions{
+		logOptions:   log.DefaultOptions(),
+		ctrlzOptions: ctrlz.DefaultOptions(),
+	}
+
+	rootCmd = &cobra.Command{
+		Use:   "Istio Webhook Controller",
+		Short: "Istio Webhook Controller",
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, args []string) {
+			if !opts.enableController {
+				log.Info("Istio Webhook Controller is not enabled, exit")
+				return
+			}
+			runWebhookController()
+		},
+	}
+)
+
+type cliOptions struct {
+	// The namespace of the webhook certificates
+	certificateNamespace string
+	kubeConfigFile       string
+	// The names of the MutatingWebhookConfiguration to manage
+	// In prototype, only one is supported.
+	mutatingWebhookConfigNames string
+	// The names of the webhooks in webhook configurations
+	// In prototype, only one is supported.
+	mutatingWebhookNames string
```

Comment:
singular

Reply:
In the code, it points to "_The names of the webhook in mutatingwebhookconfigurations, separated by comma_", hence the plural.

Reply:
If so, why not use a slice type?

Reply:
This variable only records the input parameter, which is converted to a slice at the line "mutatingWebhookNames := strings.Split(opts.mutatingWebhookNames, ",")".

Reply:
IC, have a look at `flag.StringSliceVar`